src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1053,4 +1065,79 @@ Size Pane::_GetMinSize() const
     }
 }
 
+// Event Description:
+// - Called when our control gains focus. We'll use this to trigger our GotFocus
+//   callback. The tab that's hosting us should have registered a callback which
+//   can be used to mark us as active.
+// Arguments:
+// - <unused>
+// Return Value:
+// - <none>
+void Pane::_ControlGotFocusHandler(winrt::Windows::Foundation::IInspectable const& /* sender */,
+                                   RoutedEventArgs const& /* args */)
+{
+    if (_pfnGotFocus)
+    {
+        _pfnGotFocus(shared_from_this());
+    }
+}
+
+// Method Description:
+// - Register a callback function to be called when this Pane's control gains
+//   focused. This is used by the Tab hosting us so it can track which Pane in
+//   the tree was the last one to be focused.
+// Arguments:
+// - pfnGotFocus: A function that should be called when this pane's control gets focus.
+// Return Value:
+// - <none>
+void Pane::SetGotFocusCallback(std::function<void(std::shared_ptr<Pane>)> pfnGotFocus)
+{
+    _pfnGotFocus = pfnGotFocus;

```

Comment:
```suggestion
    _pfnGotFocus = std::move(pfnGotFocus);
```

Reply:
you're taking a copy in as a param, you can elide another copy by moving here.

Reply:
Should I do the same with `Tab::SetActivePaneChangedCallback`?